pkg/runctl/controller.go

```diff
@@ -292,26 +281,22 @@ func (c *Controller) syncHandler(key string) error {
 		return nil
 	}
 
-	// Init state when undefined
-	if pipelineRun.GetStatus().State == api.StateUndefined {
-		err = pipelineRun.InitState()
-		if err != nil {
-			return err
-		}
-	}
-
 	// Check if object has deletion timestamp
 	// If not, try to add finalizer if missing
 	if pipelineRun.HasDeletionTimestamp() {
 		runManager := c.createRunManager(pipelineRun)
 		err = runManager.Cleanup(pipelineRun)
-
 		if err == nil {
-			pipelineRun.UpdateResult(api.ResultDeleted)
+			err = pipelineRun.UpdateResult(api.ResultDeleted)
+			if err != nil {
+				return err
+			}
```

Comment:
```suggestion
			pipelineRun.UpdateResult(api.ResultDeleted)
```
Update result never returns an error